File: SolvedProblems/IncreasingTripletSubsequence/IncreasingTripletSubsequence.py
def increasingTriplet(nums):
  if len(nums) < 3:
      return False
  
  iNum = float("inf")
  jNum = float("inf")

  for currentNum in nums:
      if iNum < jNum and jNum < currentNum:
          return True
      
      if currentNum > iNum:
          jNum = currentNum
      else:
          iNum = currentNum
  
  return False

increasingTriplet([1,2,3,4,5])

File: SolvedProblems/IncreasingTripletSubsequence/IncreasingTripletSubsequence.ts
function increasingTriplet(nums: number[]): boolean {
  let iNum = Infinity;
  let jNum = Infinity;

  for (let kdx = 0; kdx < nums.length; kdx++) {
    if (iNum < jNum && jNum < nums[kdx]) {
      return true;
    }

    if (nums[kdx] > iNum) {
      jNum = nums[kdx];
    } else {
      iNum = nums[kdx];
    }
  }

  return false;
}

increasingTriplet([1, 2, 3, 4, 5]); // true
